contracts/Utils.sol

```diff
@@ -23,26 +38,30 @@ contract Utils {
 
     mapping(IERC20 => uint256) internal decimals;
 
-    function getUpdateDecimals(IERC20 token) internal returns (uint256) {
-        if (token == ETH_TOKEN_ADDRESS) return ETH_DECIMALS; // save storage access
-        uint256 tokenDecimals = decimals[token];
-        // moreover, very possible that old tokens have decimals 0
-        // these tokens will just have higher gas fees.
-        if (tokenDecimals == 0) {
-            tokenDecimals = token.decimals();
+    function getSetDecimals(IERC20 token) internal returns (uint256 tokenDecimals) {
+        tokenDecimals = getDecimals(token);
```

Comment:
with this solution we are going to read storage slot 2 times.
getDecimals = 1 read (if not constant)
decimals[token] == 0 is 2nd read.
have to duplicate getDecimals code a bit to avoid double read.